pages/partials/navbar.ejs

```diff
@@ -9,140 +20,420 @@
+        displayedName = authn_user.uid;
       }
-  }
-}
-%>
+    } else {
+      displayedName = '(no user)';
+    }
+    %>
 
-<ul class="nav navbar-nav" id="username-nav">
+    <ul class="nav navbar-nav" id="username-nav">
+      <% if (devMode) { %>
+          <li class="mb-2 mb-md-0 mr-2 "><a class="btn btn-success" href="<%= urlPrefix %>/loadFromDisk">Load from disk</a></li>
+      <% } %>
 
-    <% if (!locals.effective_user_has_no_access_to_course_instance) { %>
-    <li class="nav-item dropdown mb-2 mb-md-0 mr-2 <% if (navPage == "effective") { %>active<% } %>">
-        <a class="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
-            <%= displayedName %>
-            <% if (typeof news_item_notification_count != 'undefined' && news_item_notification_count > 0) { %>
-              <span class="badge badge-pill badge-primary news-item-count"><%= news_item_notification_count %></span>
-            <% } %>
+
+      <li class="nav-item dropdown mb-2 mb-md-0 mr-2 <% if (navPage == "effective") { %>active<% } %>">
+        <a class="nav-link dropdown-toggle <% if (locals.authz_data && authz_data.overrides) { %>text-warning<% } %>" href="#" id="navbarDropdown" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+          <%= displayedName %>
+          <% if (typeof news_item_notification_count != 'undefined' && news_item_notification_count > 0) { %>
+            <span class="badge badge-pill badge-primary news-item-count"><%= news_item_notification_count %></span>
+          <% } %>
         </a>
-        <div class="dropdown-menu" aria-labelledby="navbarDropdown">
-            <a class="dropdown-item" href="<%= plainUrlPrefix %>/settings">Settings</a>
-            <% if (typeof authz_data !== 'undefined' && authz_data.authn_has_instructor_view) { %>
-            <a class="dropdown-item" href="<%= urlPrefix %>/effectiveUser">Change effective user</a>
-            <% } %>
-            <a class="dropdown-item news-item-link" href="<%= urlPrefix %>/news_items">
-              News
-              <% if (typeof news_item_notification_count != 'undefined' && news_item_notification_count > 0) { %>
-                <span class="badge badge-pill badge-primary news-item-link-count"><%= news_item_notification_count %></span>
+        <div class="dropdown-menu dropdown-menu-right" aria-labelledby="navbarDropdown">
+
+          <% if (locals.authn_is_administrator) { %>
+
+            <button class="dropdown-item" type="button" name="administrator">
+              <% if (locals.access_as_administrator) { %>Turn off administrator access<% } else { %>Turn on administrator access<% } %>
+            </button>
+
+            <div class="dropdown-divider"></div>
+
+          <% } %>
+
+          <% if (locals.course_instance && (authz_data.authn_has_course_permission_preview || authz_data.authn_has_course_instance_permission_view)) {
```

Comment:
I felt it was better to keep this in the same menu, because changes to "View type" also involve changes to "Effective user". What you propose is certainly an option - I'd be happier leaving it to a future PR, if it's clear that doing so will be an improvement.

Reply:
As I commented above, I'm wondering whether we can separate view type from effective user.

The main UI issue I'm worried about it is making it clear to users which view they are in, and where to go to change the view. At the moment we have these huge obnoxious Instructor/Student buttons in the navbar. I like getting rid of those, but I fear we might have gone too far in hiding everything under the user's name with no other indications of what's going on. Maybe we could just change the navbar item to be "User Name (instructor)" where the "(instructor)" text changes to reflect the view type?